File: HW/HW11/hw11.py
class HW11:
    def floodFill(self, image, sr, sc, color):
        oldColor = image[sr][sc]

        def depth_first(x, y):
            if x < 0 or y < 0 or x >= len(image) or y >= len(image[0]) or image[x][y] == color or image[x][y] != oldColor:
                return
            image[x][y] = color
            depth_first(x + 1, y)
            depth_first(x - 1, y)
            depth_first(x, y + 1)
            depth_first(x, y - 1)

        depth_first(sr, sc)

        return image


hw11 = HW11()
image = [[1, 1, 1], [1, 1, 0], [1, 0, 1]]
print(hw11.floodFill(image, 1, 1, 2))
image = [[0, 0, 0], [0, 0, 0]]
print(hw11.floodFill(image, 0, 0, 0))

File: HW/HW11/hw11.java
package hw11;

import java.io.Console;
import java.util.Arrays;

public class hw11 {
    public static int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int oldColor = image[sr][sc];
        depthFirst(sr, sc, oldColor, color, image);
        return image;
    }

    public static void depthFirst(int x, int y, int oldColor, int newColor, int[][] image){
        if  (x < 0 || y < 0 || x >= image.length || y >= image[0].length || image[x][y] == newColor || image[x][y] != oldColor)
            return;
        image[x][y] = newColor;
        depthFirst(x - 1, y, oldColor, newColor, image);
        depthFirst(x + 1, y, oldColor, newColor, image);
        depthFirst(x, y + 1, oldColor, newColor, image);
        depthFirst(x, y - 1, oldColor, newColor, image);
    }

    public static void main(String[] args) {
        int[][] image = {
                {1, 1, 1},
                {1, 1, 0},
                {1, 0, 1}
        };
        System.out.println(Arrays.deepToString(floodFill(image, 1, 1, 2)));
        image = new int[][]{
                {0, 0, 0},
                {0, 0, 0},
        };
        System.out.println(Arrays.deepToString(floodFill(image, 0, 0, 0)));
    }
}